src/rust/engine/fs/src/lib.rs

```diff
@@ -463,25 +463,26 @@ pub struct PathGlobs {
 
 impl PathGlobs {
   pub fn create(
-    include: &[String],
-    exclude: &[String],
+    globs: &[String],
     strict_match_behavior: StrictGlobMatching,
     conjunction: GlobExpansionConjunction,
   ) -> Result<PathGlobs, String> {
-    let include = PathGlob::spread_filespecs(include)?;
-    Self::create_with_globs_and_match_behavior(include, exclude, strict_match_behavior, conjunction)
-  }
-
-  fn create_with_globs_and_match_behavior(
-    include: Vec<PathGlobIncludeEntry>,
-    exclude: &[String],
-    strict_match_behavior: StrictGlobMatching,
-    conjunction: GlobExpansionConjunction,
-  ) -> Result<PathGlobs, String> {
-    let gitignore_excludes = GitignoreStyleExcludes::create(exclude)?;
+    // NB: We use a loop, rather than `.filter()`, to avoid traversing the globs twice.
```

Comment:
We could also consider using `.partition()` here: https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.partition

Reply:
Neat! This is exactly what Greg and I were looking for! I don't think it will work, though, specifically because of this line:
```rust
let normalized_exclude: String = glob.chars().skip(1).collect();
```

We must map the excludes, but only the excludes.

But great to know for the future! cc @gshuflin